internal/services/storage/storage_account_resource.go

```diff
@@ -2039,9 +2039,7 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 		blobClient := storageClient.BlobServicesClient
 		blobProps, err := blobClient.GetServiceProperties(ctx, id.ResourceGroup, id.Name)
 		if err != nil {
-			if !utils.ResponseWasNotFound(blobProps.Response) {
-				return fmt.Errorf("reading blob properties for AzureRM Storage Account %q: %+v", id.Name, err)
-			}
+			return fmt.Errorf("reading blob properties for AzureRM Storage Account %q: %+v", id.Name, err)
```

Comment:
from memory this API can only be called for some kind of Storage Accounts, otherwise it returns a 404 - so I think this'll break those (I want to say Blob/File accounts?)

Reply:
Then we shall accordingly update the code that evaluates the `supportLevel`?https://github.com/hashicorp/terraform-provider-azurerm/blob/ce372de3d155e408da9f9d4df8f3bda0bb9c1c22/internal/services/storage/storage_account_resource.go#L55

Reply:
@tombuildsstuff Yes, I've tried to deploy storage accounts of all the five possible `account_kind`, which are all successfully created and have no diff in a follow-up plan. I've checked that these possible `account_kind` are also covered in the existing acctests.